src/components/Menu/MenuItem.tsx

```diff
@@ -176,6 +272,48 @@ class MenuItem extends UIComponent<Extendable<MenuItemProps>, MenuItemState> {
 
     _.invoke(this.props, 'onFocus', e, this.props)
   }
+
+  private closeMenu = e => {
+    const { menu } = this.props
+    const { submenuOpen } = this.state
+    if (menu && submenuOpen) {
+      this.setState({ submenuOpen: false }, () => focusAsync(this.itemRef.current))
+    }
+  }
+
+  private closeSubmenu = e => {
+    const { menu } = this.props
+    const { submenuOpen } = this.state
+    if (menu && submenuOpen) {
+      this.setState({ submenuOpen: false }, () => focusAsync(this.itemRef.current))
+      e.stopPropagation()
+    }
+  }
+
+  private openSubmenu = e => {
+    const { menu } = this.props
+    const { submenuOpen } = this.state
+    if (menu && !submenuOpen) {
+      this.handleClick(e)
+      e.stopPropagation()
+      e.preventDefault()
+    }
+  }
+
+  private handleSubmenuClicked = (e: React.SyntheticEvent, props: MenuProps) => {
+    this.trySetState({ submenuOpen: false })
+    _.invoke(props, 'onClick', props)
+  }
+
+  private handleSubmenuKeyDown = (e: React.SyntheticEvent, props: MenuProps) => {
+    if (
+      keyboardKey.getCode(e) === keyboardKey.Enter ||
+      keyboardKey.getCode(e) === keyboardKey.Spacebar
+    ) {
+      this.trySetState({ submenuOpen: false })
+    }
+    _.invoke(props, 'onKeyDown', props)
+  }
```

Comment:
Keys should be assigned to actions on behaviors. Also I think we can one action called ```performSubmenuClick``` which will call ```handleSubmenuClicked``` on Enter or Space. Same way as we have on MenuItem ```performClick``` line 231.

So basically in ```menuItemBehavior``` you add to key action

```js
keyActions: {
   ...
   
   submenu: {
      performSubmenuClick: {
        keyCombinations: [{ keyCode: keyboardKey.Enter }, { keyCode: keyboardKey.Spacebar }],
      },
    },
   ...
}
```

In ```MenuItem``` comp, add action handler (line 231)

```js
actionHandlers = {
    ...
    performSubmenuClick: (event) => this.handleSubmenuClicked(event, null)

    ...
}
```

NOTE: I see that in your handlers you receive props as second param. In case of ```keydown``` event, client code is handled in```getKeyDownHandlers.ts```, by calling ```_.invoke(props, 'onKeyDown', event)```, so we can just pass NULL as 2nd param in action handler.

and then line 188 where you create submenu

```js
const maybeSubmenu =
       ...
        Menu.create(menu, {
            defaultProps: {
              accessibility: menuBehavior, // I don't see any reason to have submenuBehavior for now as they are the same
              
            },
            overrideProps: {
              onClick: this.handleSubmenuClicked,
             ...accessibility.keyHandlers.submenu   // adds onkeyDown handler which will call 'performSubmenuClick'
            },
          })
        : null
```

Please let me know whta you think.